From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nopjmp <kthompson@hey.com>
Date: Fri, 12 Mar 2021 20:52:00 -0600
Subject: [PATCH] Configurable Player Data Removal

This allows the server operator to remove player information from the logs.

This is currently enabled by default since many anarchy server owners want this.

diff --git a/src/main/java/dev/pomf/dionysus/DionysusConfig.java b/src/main/java/dev/pomf/dionysus/DionysusConfig.java
index e3ca26e90e956cca655f3449e3985079cc842f46..81f24b09bfc127226918aea6215a79022307d8af 100644
--- a/src/main/java/dev/pomf/dionysus/DionysusConfig.java
+++ b/src/main/java/dev/pomf/dionysus/DionysusConfig.java
@@ -222,4 +222,12 @@ public class DionysusConfig {
             Bukkit.getLogger().info("Fair natural spawns enabled, using the Spigot world mob-spawn-range for search and Bukkit spawn-limits for the count");
         }
     }
+
+    public static boolean redactPlayerData = true;
+    private static void redactPlayerData() {
+        redactPlayerData = getBoolean("redact-player-data", redactPlayerData);
+        if (redactPlayerData) {
+            Bukkit.getLogger().info("Redact player data enabled, IP Address and location data will be removed.");
+        }
+    }
 }
diff --git a/src/main/java/net/minecraft/server/LoginListener.java b/src/main/java/net/minecraft/server/LoginListener.java
index 06353e7d312121c18a942f7fe6f31d11efcc8b01..d26c1798b64333b3e1c04e72ea1823ed453bae1e 100644
--- a/src/main/java/net/minecraft/server/LoginListener.java
+++ b/src/main/java/net/minecraft/server/LoginListener.java
@@ -4,6 +4,7 @@ import com.destroystokyo.paper.profile.CraftPlayerProfile;
 import com.destroystokyo.paper.profile.PlayerProfile;
 import com.mojang.authlib.GameProfile;
 import com.mojang.authlib.exceptions.AuthenticationUnavailableException;
+import dev.pomf.dionysus.DionysusConfig;
 import io.netty.channel.ChannelFuture;
 import io.netty.channel.ChannelFutureListener;
 import io.netty.util.concurrent.Future;
@@ -186,7 +187,11 @@ public class LoginListener implements PacketLoginInListener, ITickable {
     }
 
     public String c() {
-        return this.i != null ? this.i + " (" + this.networkManager.getSocketAddress() + ")" : String.valueOf(this.networkManager.getSocketAddress());
+        String address = this.networkManager.getSocketAddress().toString();
+        if (DionysusConfig.redactPlayerData) {
+            address = "/0.0.0.0:0";
+        }
+        return this.i != null ? this.i + " (" + address + ")" : address;
     }
 
     public void a(PacketLoginInStart packetlogininstart) {
diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index abcd062371372f07b00a732ca4cf479491d961e3..99a803bc45a18420fb3e30e35ad1eb94e260ce6f 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -4,6 +4,7 @@ import com.google.common.collect.Lists;
 import com.google.common.primitives.Doubles;
 import com.google.common.primitives.Floats;
 import com.google.common.util.concurrent.Futures;
+import dev.pomf.dionysus.DionysusConfig;
 import io.netty.util.concurrent.Future;
 import io.netty.util.concurrent.GenericFutureListener;
 import java.io.IOException;
@@ -363,7 +364,11 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
 
                 if (d10 - d9 > Math.max(100.0D, Math.pow(org.spigotmc.SpigotConfig.movedTooQuicklyMultiplier * (float) i * speed, 2)) && (!this.minecraftServer.R() || !this.minecraftServer.Q().equals(entity.getName()))) { // Spigot
                 // CraftBukkit end
-                    PlayerConnection.LOGGER.trace("{} (vehicle of {}) moved too quickly! {},{},{}", entity.getName(), this.player.getName(), Double.valueOf(d6), Double.valueOf(d7), Double.valueOf(d8));
+                    if (DionysusConfig.redactPlayerData) {
+                        PlayerConnection.LOGGER.trace("{} (vehicle of {}) moved too quickly!", entity.getName(), this.player.getName());
+                    } else {
+                        PlayerConnection.LOGGER.trace("{} (vehicle of {}) moved too quickly! {},{},{}", entity.getName(), this.player.getName(), d6, d7, d8);
+                    }
                     this.networkManager.sendPacket(new PacketPlayOutVehicleMove(entity));
                     return;
                 }
@@ -611,7 +616,11 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
 
                                 if (d11 - d10 > Math.max(f2, Math.pow(org.spigotmc.SpigotConfig.movedTooQuicklyMultiplier * (float) i * speed, 2)) && (!this.minecraftServer.R() || !this.minecraftServer.Q().equals(this.player.getName()))) { // Spigot
                                 // CraftBukkit end
-                                    PlayerConnection.LOGGER.trace("{} moved too quickly! {},{},{}", this.player.getName(), Double.valueOf(d7), Double.valueOf(d8), Double.valueOf(d9));
+                                    if (DionysusConfig.redactPlayerData) {
+                                        PlayerConnection.LOGGER.trace("{} moved too quickly!", this.player.getName());
+                                    } else {
+                                        PlayerConnection.LOGGER.trace("{} moved too quickly! {},{},{}", this.player.getName(), d7, d8, d9);
+                                    }
                                     this.a(this.player.locX, this.player.locY, this.player.locZ, this.player.yaw, this.player.pitch);
                                     return;
                                 }
diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
index df7b6d3ee9831b85f498e64a1117a41a384d02b3..2b54448a8846402625e6df5ba56925eceaa440b1 100644
--- a/src/main/java/net/minecraft/server/PlayerList.java
+++ b/src/main/java/net/minecraft/server/PlayerList.java
@@ -5,6 +5,7 @@ import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import com.mojang.authlib.GameProfile;
+import dev.pomf.dionysus.DionysusConfig;
 import io.netty.buffer.Unpooled;
 import java.io.File;
 import java.net.SocketAddress;
@@ -292,7 +293,12 @@ public abstract class PlayerList {
         }
         // Paper end
         // CraftBukkit - Moved from above, added world
-        PlayerList.f.info(entityplayer.getName() + "[" + s1 + "] logged in with entity id " + entityplayer.getId() + " at ([" + entityplayer.world.worldData.getName() + "]" + entityplayer.locX + ", " + entityplayer.locY + ", " + entityplayer.locZ + ")");
+        // Dionysus - anonymize logs
+        if (DionysusConfig.redactPlayerData) {
+            PlayerList.f.info(entityplayer.getName() + "[/0.0.0.0:0] logged in with entity id " + entityplayer.getId() + " at ([REDACTED] 0.0, 0.0, 0.0)");
+        } else {
+            PlayerList.f.info(entityplayer.getName() + "[" + s1 + "] logged in with entity id " + entityplayer.getId() + " at ([" + entityplayer.world.worldData.getName() + "]" + entityplayer.locX + ", " + entityplayer.locY + ", " + entityplayer.locZ + ")");
+        }
     }
 
     public void sendScoreboard(ScoreboardServer scoreboardserver, EntityPlayer entityplayer) {
diff --git a/src/main/java/net/minecraft/server/ServerConnection.java b/src/main/java/net/minecraft/server/ServerConnection.java
index c5a2aad1c19993e9903cfc6f0decb85a3b68cebf..87722b33a084f95a028a091c4318431505fee040 100644
--- a/src/main/java/net/minecraft/server/ServerConnection.java
+++ b/src/main/java/net/minecraft/server/ServerConnection.java
@@ -2,6 +2,7 @@ package net.minecraft.server;
 
 import com.google.common.collect.Lists;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import dev.pomf.dionysus.DionysusConfig;
 import dev.pomf.dionysus.util.random.Xorshift;
 import io.netty.bootstrap.ServerBootstrap;
 import io.netty.channel.Channel;
@@ -178,7 +179,11 @@ public class ServerConnection {
                                 throw new ReportedException(crashreport);
                             }
 
-                            ServerConnection.e.warn("Failed to handle packet for {}", networkmanager.getSocketAddress(), exception);
+                            String address = networkmanager.getSocketAddress().toString();
+                            if (DionysusConfig.redactPlayerData) {
+                                address = "/0.0.0.0:0";
+                            }
+                            ServerConnection.e.warn("Failed to handle packet for {}", address, exception);
                             final ChatComponentText chatcomponenttext = new ChatComponentText("Internal server error");
 
                             networkmanager.sendPacket(new PacketPlayOutKickDisconnect(chatcomponenttext), new GenericFutureListener() {
